stream.go

```diff
@@ -0,0 +1,78 @@
+package gogpt
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"log"
+	"net/http"
+)
+
+// CreateCompletionStream — API call to create a completion w/ streaming
+// support. It sets whether to stream back partial progress. If set, tokens will be
+// sent as data-only server-sent events as they become available, with the
+// stream terminated by a data: [DONE] message.
+func (c *Client) CreateCompletionStream(
+	ctx context.Context,
+	request CompletionRequest,
+) ([]CompletionResponse, error) {
+	request.Stream = true
+	reqBytes, err := json.Marshal(request)
+	if err != nil {
+		return nil, err
+	}
+
+	urlSuffix := "/completions"
+	req, err := http.NewRequest("POST", c.fullURL(urlSuffix), bytes.NewBuffer(reqBytes))
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", "text/event-stream")
+	req.Header.Set("Cache-Control", "no-cache")
+	req.Header.Set("Connection", "keep-alive")
+	if err != nil {
+		return nil, err
+	}
+
+	req = req.WithContext(ctx)
+	res, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	reader := bufio.NewReader(res.Body)
+	var line []byte
+	var headerData = []byte("data: ")
+
+	var responses []CompletionResponse
+	for {
```

Comment:
This loop effectively makes it a non-streaming API. I suggest this method returns a struct like:

```go

type CompletionStream struct {
	reader *bufio.Reader
	response *http.Response
}

func (stream *CompletionStream) Recv() {
	// do the loop body logic with reader.ReadBytes, etc.
}

func (stream *CompletionStream) Close() {
	// close response body, reader, etc. (basically what you've done in defers)
}

func (c *Client) CreateCompletionStream(
	ctx context.Context,
	request CompletionRequest,
) (*CompletionStream, error) {
	// make request and construct stream, return before the loop
}
```

Reply:
Very well. I will iterate and re-submit. Thank you for the guidance.